File: app-android/app/src/main/java/de/tutao/tutanota/credentials/DataKeyGenerator.java
package de.tutao.tutanota.credentials;

import javax.crypto.SecretKey;

/**
 * Interface for obtaining keys that can be used to encrypt arbitrary data using android's keystore API.
 */
public interface DataKeyGenerator {
	SecretKey generateDataKey(String alias, CredentialEncryptionMode credentialEncryptionMode);
}

File: app-android/tutashared/src/main/java/de/tutao/tutashared/credentials/DataKeyGenerator.kt
package de.tutao.tutashared.credentials

import javax.crypto.SecretKey

/**
 * Interface for obtaining keys that can be used to encrypt arbitrary data using android's keystore API.
 */
interface DataKeyGenerator {
	fun generateDataKey(alias: String, credentialEncryptionMode: CredentialEncryptionMode): SecretKey
}